### PR TITLE
Clean up BTDataCollector

### PR DESCRIPTION
**Pull request by scannillo**
### Summary

- Addresses comments in #595 
- Android (and JS to my understanding) doesn't have an equivalent of `BTDataCollectorDelegate` for merchants to hook into notifications for when Kount started, succeeded, or errored.
- We can remove `collectCardFraudData` in favor of `collectDeviceData`, since both methods collected all data (Kount & PayPal). This also matches Android.

### Changes

- Remove `BTDataCollectorDelegate`
- Remove `BTDataCollector.collectCardFraudData()` method
- Remove `BTDataCollectorKountErrorDomain`

### Checklist

- [X] Added a changelog entry

### Authors

- @scannillo

### CODE REVIEW

File: UnitTests/BraintreeDataCollectorTests/BTDataCollector_Tests.swift [hunk 1]
```diff
@@ -56,12 +28,11 @@ class BTDataCollector_Tests: XCTestCase {
         
         waitForExpectations(timeout: 2, handler: nil)
     }
-    
-    func testCollectDeviceDataWithCompletionBlock_whenMerchantHasKountConfiguration_usesConfiguration() {
+
+    func testCollectDeviceData_whenMerchantConfiguredForKount_collectsAllData() {
```
Comment (scannillo) [inline UnitTests/BraintreeDataCollectorTests/BTDataCollector_Tests.swift:32]: Renamed these test names to be more clear

File: V5_MIGRATION.md [hunk 2]
```diff
@@ -171,3 +171,9 @@ venmoDriver.tokenizeVenmoAccount(with: venmoRequest) { (venmoAccountNonce, error
   // transact with nonce on server
 }
 ```
+
+## BraintreeDataCollector
```
Comment (sestevens) [inline V5_MIGRATION.md:175]:
```suggestion
## Data Collector
```

The other sections are named more generically (e.g., `Card`, `Venmo`) so maybe we should do the same here.

**Review by sestevens — APPROVED** [review]
Looks good! Just one small suggestion about the migration guide.

File: CHANGELOG.md [hunk 3]
```diff
@@ -34,6 +34,9 @@
     * `BTVenmoDriver.authorizeAccount(profileID:vault:completion:)`
   * Remove `initWithNumber` and `initWithParameters` initializers from `BTCard`
   * Replace `BTVenmoDriver.authorizeAccount` methods with `BTVenmoDriver.tokenizeVenmoAccount`
+  * Remove `BTDataCollectorDelegate`
```
Comment (hollabaq86) [inline CHANGELOG.md:37]: Wonder if we should add a subsection for data collector updates? similar to what we did for PP updates in #612

**Review by sshropshire — APPROVED** [review]
🌊

**Review by hollabaq86 — APPROVED** [review]
LGTM :shipit:

File: CHANGELOG.md (hunk 3 above)
Comment (scannillo) [reply CHANGELOG.md]: Good call - DONE!

**Review by sestevens — APPROVED** [review]
🌟

**Pull request merged**